src/Particle/createDistanceTableAB.cpp

```
@@ -25,14 +25,14 @@ namespace qmcplusplus
  *\param s source/target particle set
  *\return index of the distance table with the name
  */
-DistanceTableData* createDistanceTableAB(const ParticleSet& s, ParticleSet& t, std::ostream& description)
+DistanceTable* createDistanceTableAB(const ParticleSet& s, ParticleSet& t, std::ostream& description)
```

why not a smart pointer?

These need to throw an exception if they fail, when it happens it will frequently be in another classes a constructor.

---

I also caught this but consider this a follow-up topic as more places needs to be touched.

---

Added #3616 to remember it.